### PR TITLE
feat(botocore): support sending LLMObs payloads

### PR DESCRIPTION
**Pull request by Yun-Kim**
This PR introduces LLMObs payloads support from the botocore/bedrock integration, and introduces the following environment variable feature flags:
- `DD_BEDROCK_LLMOBS_ENABLED`: enables sending llmobs payloads
- `DD_LANGCHAIN_LLMOBS_PROMPT_COMPLETION_SAMPLE_RATE`: controls llmobs record sample rate, default is 1.0 (100%)
- Reuses `DD_API_KEY` and `DD_APP_KEY`

The following record payload is being sent to LLMObs ingest API:
Screenshot 2024-01-12 at 10 45 07 AM

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by sabrenner — APPROVED** [review]
LGTM!

**Review by mabdinur — APPROVED** [review]
Let's merge this after the bedrock integration

File: ddtrace/contrib/botocore/services/bedrock.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    def __iter__(self):
+        try:
+            for line in self.__wrapped__:
+                self._body.append(json.loads(line["chunk"]["bytes"]))
+                yield line
+            metadata = _extract_streamed_response_metadata(self._datadog_span, self._body)
+            formatted_response = _extract_streamed_response(self._datadog_span, self._body)
+            self._process_response(formatted_response, metadata=metadata)
+            self._datadog_span.finish()
+        except Exception:
```
[the remark this answers is not in the document]
Comment (Yun-Kim) [reply ddtrace/contrib/botocore/services/bedrock.py:87]: Thanks for the catch!

File: ddtrace/internal/llmobs/integrations/bedrock.py [hunk 2]
```diff
@@ -1,5 +1,80 @@
...
+    @classmethod
+    def _llmobs_tags(cls, span: Span) -> List[str]:
+        tags = [
+            "version:%s" % (config.version or ""),
+            "env:%s" % (config.env or ""),
+            "service:%s" % (span.service or ""),
+            "src:integration",
+            "ml_obs.request.model:%s" % (span.get_tag("bedrock.request.model") or ""),
+            "ml_obs.request.model_provider:%s" % (span.get_tag("bedrock.request.model_provider") or ""),
+            "ml_obs.request.error:%d" % span.error,
```
Comment (Yun-Kim) [inline ddtrace/internal/llmobs/integrations/bedrock.py:26]: @gary-huang

**Pull request merged**